src/platform/wasm.zig

```diff
@@ -117,38 +134,84 @@ pub const Platform = struct {
         return js.machHasEvent();
     }
 
-    pub fn pollEvent(_: *Platform) ?structs.Event {
+    pub fn pollEvent(platform: *Platform) ?structs.Event {
         const event_type = js.machEventShift();
 
         return switch (event_type) {
-            1 => structs.Event{
-                .key_press = .{ .key = @intToEnum(enums.Key, js.machEventShift()) },
+            1 => key_press: {
+                const key = @intToEnum(enums.Key, js.machEventShift());
+                switch (key) {
+                    .left_shift, .right_shift => platform.last_key_mods.shift = true,
+                    .left_control, .right_control => platform.last_key_mods.control = true,
+                    .left_alt, .right_alt => platform.last_key_mods.alt = true,
+                    .left_super, .right_super => platform.last_key_mods.super = true,
+                    .caps_lock => platform.last_key_mods.caps_lock = true,
+                    .num_lock => platform.last_key_mods.num_lock = true,
+                    else => {},
+                }
+                break :key_press structs.Event{
+                    .key_press = .{
+                        .key = key,
+                        .mods = platform.last_key_mods,
+                    },
+                };
             },
-            2 => structs.Event{
-                .key_release = .{ .key = @intToEnum(enums.Key, js.machEventShift()) },
```

Comment:
This switch now no longer handles 2, it jumps from 1 to 3. Why is that?

Reply:
I didn't add js code to handle key_repeat (same reason as above). This one is a pretty trivial change though so probably ok.